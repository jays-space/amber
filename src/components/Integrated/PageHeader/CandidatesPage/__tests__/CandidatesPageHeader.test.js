import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { CandidatesPageHeader } from "..";
import { BrowserRouter } from "react-router-dom";

const HeaderWithMockedRouter = ({ noOfCandidates }) => {
  return (
    <BrowserRouter>
      <CandidatesPageHeader noOfCandidates={noOfCandidates} />
    </BrowserRouter>
  );
};

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("<CandidatesPageHeader />", () => {
  it("should render correctly", () => {
    render(<HeaderWithMockedRouter noOfCandidates={1} />);

    const headerComponent = screen.getByTestId("candidates-page-header");
    expect(headerComponent).toBeInTheDocument();
    
    const headingElement = screen.getByRole("heading", { name: "1 candidate" });
    expect(headingElement).toBeInTheDocument();
    expect(headingElement).toMatchSnapshot();
  });

  it("should render candidate count greater tha one", () => {
    render(<HeaderWithMockedRouter noOfCandidates={10} />);

    const headingElement = screen.getByRole("heading", { name: "10 candidates" });
    expect(headingElement).toBeInTheDocument();
  });
});
