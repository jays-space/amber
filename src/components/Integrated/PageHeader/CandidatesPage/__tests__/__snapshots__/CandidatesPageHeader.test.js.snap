// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CandidatesPageHeader /> should render correctly 1`] = `
<h1
  class="mr-10 text-slate-700 capitalize text-2xl lg:text-4xl font-semibold"
>
  1 candidate
</h1>
`;
